Harmonize user docs

NobodysNightmare opened the pull request:
## OpenProject work package
- https://community.openproject.org/work_packages/18094
## Remark

I was brave enough to **implement** the missing link to the already existing delete action.
Note that I am a ruby newbie and my local setup does not allow me to run the tests... So you should be really careful reviewing my code changes ;-)

NobodysNightmare committed:
fix table alignment

[ci skip]

NobodysNightmare committed:
harmonize user spec

- add linked properties
- specify status and its values
- also add delete action link to actions

[ci skip]

NobodysNightmare committed:
harmonize user examples

[ci skip]

NobodysNightmare committed:
actually include delete link

how hard can ruby be anyway...

I will for certain regret that commit soon ^^

TeatroIO commented:
I've prepared a stage. <a href='http://feature-harmonize-user-docs.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Click to open.</a>

NobodysNightmare committed:
obey style rules

NobodysNightmare committed:
fix reference to current user

NobodysNightmare committed:
test for missing delete link

NobodysNightmare committed:
actually call the method I intended

apparently I am not able to distinguish a class method from an instance
method ^^

NobodysNightmare committed:
remove param from method

NobodysNightmare committed:
fix tests, use mocking

ulferts commented on doc/apiv3-documentation.apib:
```diff
@@ -1991,9 +2003,9 @@ This endpoint lists the types that are *available* in a given project.
                 "lastName": "Sheppard",
                 "mail": "shep@mail.com",
                 "avatar": "https://gravatar/avatar",
+				"status": "active",
```
indentation

NobodysNightmare replied:
Thanks for that one... I managed to introduce >1000 tabs with this PR ^^

Editor settings are now fixed again.

ulferts commented on doc/apiv3-documentation.apib:
```diff
@@ -1958,9 +1964,15 @@ This endpoint lists the types that are *available* in a given project.
+
+* `active` - the user can act normally
+* `registered` - the user just registered to the instance, activation by an administrator is required
+* `locked` - the user was locked and can't login
```
Just some questions regarding wording here:
- what do you mean by "can act normally"? I wouldn't want to use the word "normally" to describe my behaviour at the best of times.
  - Proposal: "The user can log in with the account" 
- `activation by an administrator is required`. He could also be in the first phase of registering himself. He would then have to activate his account by following the link in the e-mail.
  - Proposal: "The user can not log in yet but will be able to once the registration is completed."
- `the user was locked and can't login` the account is still locked so "was" seems wrong to me.

ulferts commented:
LGTM except for the minor things I annotated.

NobodysNightmare committed:
fix indentation

[ci skip]

NobodysNightmare committed:
improve wording

[ci skip]
